flytekit/control_plane/nodes.py

```diff
@@ -279,18 +275,47 @@ def promote_from_model(cls, base_model: _node_execution_models.NodeExecution) ->
             closure=base_model.closure, id=base_model.id, input_uri=base_model.input_uri, metadata=base_model.metadata
         )
 
+    def get_interface(self) -> "flytekit.control_plane.interface.TypedInterface":
+        """
+        Return the interface of the task or subworkflow associated with this node execution.
+        """
+        from flytekit.control_plane.tasks.task import FlyteTask
+        from flytekit.control_plane.workflow import FlyteWorkflow
+
+        if not self.metadata.is_parent_node:
+            # if not a parent node, assume a task execution node
+            task_id = self.task_executions[0].id.task_id
+            task = FlyteTask.fetch(task_id.project, task_id.domain, task_id.name, task_id.version)
+            return task.interface
+
+        # otherwise assume the node is associated with a subworkflow
+        client = _flyte_engine.get_client()
+        lp_id = client.get_execution(self.id.execution_id).spec.launch_plan
+        workflow = FlyteWorkflow.fetch(lp_id.project, lp_id.domain, lp_id.name, lp_id.version)
+        flyte_subworkflow_node: FlyteNode = [n for n in workflow.nodes if n.id == self.id.node_id][0]
+        return flyte_subworkflow_node.target.flyte_workflow.interface
+
     def sync(self):
         """
         Syncs the state of the underlying execution artifact with the state observed by the platform.
         """
-        if not self.is_complete or self.task_executions is not None:
+        if not self.is_complete or self._task_executions is None:
```

Comment:
should you also check if is_parent_node is false here too?

Reply:
updated this to account for parent node